### ISSUE TITLE
Query to DataStore return invalid_grant error

### ISSUE THREAD

**Issue by isuhendro** [GoogleCloudPlatform/google-cloud-node#2306]
When my NodeJS app `run` query to Google DataStore, it works well in my local machine.
I am using this npm `@google-cloud/datastore` library

However, once I create docker image, and run the same application in `docker`, the same `run` command returns below error
```
{ '0': { [Error: invalid_grant] code: 401, metadata: Metadata { _internal_repr: {} } } }
Auth error:Error: invalid_grant
E0513 02:52:21.261033396       5 plugin_credentials.c:74]    Getting metadata from plugin failed with error: invalid_grant
```
Would you be able to advise how to fix / investigate? Thanks!

**Comment by stephenplusplus** [comment]
Sorry to hear it's being a problem! This isn't an error we see often, so it might take us a few tries to sort it out. To get the basics out of the way:

- how are you instantiating the Datastore module?
- are you trying to connect to a custom endpoint, or are you using the remote Datastore API?
- are you authenticating with a keyfile, a credentials JSON object, something else?

**Comment by isuhendro** [comment]
@stephenplusplus thanks a lot for responding. It appears the error is my bad.
This was due to in `Dockerfile` I specified
```
EXPOSE 5555
```
but while running the docker I am using
```
docker run --rm -p 2222:2222 gcr.io/${PROJECT_ID}/myapp:v1
```
Once I changed `5555` to `2222` in DockerFile, it works.

I still could not understand why _only_ accessing Google DataStore is impacted with my careless mistake above though. Thanks again @stephenplusplus
